File: py.easy/math.7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#

# @lc code=start
import math
class Solution:
    def reverse(self, x: int) -> int:
        result = 0
        while(x != 0):
            if (x >= 0):
                first = x % 10
            else:
                first = x % 10 - 10
                if (first == -10):
                    first = 0
            x = math.trunc(x/10)
            result = result*10 + first
        MIN = -2**31
        MAX = 2**31-1
        return result if result > MIN and result < MAX else 0
# @lc code=end

File: js.easy/math.7.整数反转.js
/*
 * @lc app=leetcode.cn id=7 lang=javascript
 *
 * [7] 整数反转
 */

// @lc code=start
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function (x) {
  const MIN = Math.pow(-2, 31);
	const MAX = Math.pow(2, 31) - 1;
	if (x == 0) {
		return 0;
	}
	let first = x;
	let result = 0;
	while (x / 10) {
		first = x % 10;
		x = parseInt(x / 10);
		result = result * 10 + first;
	}
	return (result>MIN&&result<MAX)?result:0;
};
// @lc code=end

